lte/gateway/c/oai/tasks/s1ap/s1ap_mme_encoder.c

```diff
@@ -19,437 +19,142 @@
-
-static inline int s1ap_mme_encode_e_rab_release_command(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length);
-//------------------------------------------------------------------------------
-static inline int s1ap_mme_encode_paging(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length);
-
-static inline int s1ap_mme_encode_ue_context_modification_request(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length);
-
-static inline int s1ap_mme_encode_mme_configuration_transfer(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length);
-
-static inline int s1ap_mme_encode_pathswitchreqack(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length);
-
-static inline int s1ap_mme_encode_pathswitchreqfailure(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length);
-
-static inline int s1ap_mme_encode_initial_context_setup_request(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length) {
-  S1ap_InitialContextSetupRequest_t initialContextSetupRequest;
-  S1ap_InitialContextSetupRequest_t* initialContextSetupRequest_p =
-      &initialContextSetupRequest;
-
-  memset(
-      initialContextSetupRequest_p, 0,
-      sizeof(S1ap_InitialContextSetupRequest_t));
-
-  if (s1ap_encode_s1ap_initialcontextsetuprequesties(
-          initialContextSetupRequest_p,
-          &message_p->msg.s1ap_InitialContextSetupRequestIEs) < 0) {
-    return -1;
-  }
-
-  return s1ap_generate_initiating_message(
-      buffer, length, S1ap_ProcedureCode_id_InitialContextSetup,
-      S1ap_Criticality_reject, &asn_DEF_S1ap_InitialContextSetupRequest,
-      initialContextSetupRequest_p);
-}
-
+    S1ap_S1AP_PDU_t* pdu, uint8_t** buffer, uint32_t* len);
 //------------------------------------------------------------------------------
 int s1ap_mme_encode_pdu(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length) {
-  DevAssert(message_p != NULL);
+    S1ap_S1AP_PDU_t* pdu, uint8_t** buffer, uint32_t* length) {
+  int ret = -1;
+  DevAssert(pdu != NULL);
   DevAssert(buffer != NULL);
   DevAssert(length != NULL);
 
-  switch (message_p->direction) {
-    case S1AP_PDU_PR_initiatingMessage:
-      return s1ap_mme_encode_initiating(message_p, buffer, length);
+  switch (pdu->present) {
+    case S1ap_S1AP_PDU_PR_initiatingMessage:
+      ret = s1ap_mme_encode_initiating(pdu, buffer, length);
+      break;
 
-    case S1AP_PDU_PR_successfulOutcome:
-      return s1ap_mme_encode_successfull_outcome(message_p, buffer, length);
+    case S1ap_S1AP_PDU_PR_successfulOutcome:
+      ret = s1ap_mme_encode_successfull_outcome(pdu, buffer, length);
+      break;
 
-    case S1AP_PDU_PR_unsuccessfulOutcome:
-      return s1ap_mme_encode_unsuccessfull_outcome(message_p, buffer, length);
+    case S1ap_S1AP_PDU_PR_unsuccessfulOutcome:
+      ret = s1ap_mme_encode_unsuccessfull_outcome(pdu, buffer, length);
+      break;
 
     default:
-      OAILOG_DEBUG(
+      OAILOG_NOTICE(
```

Comment:
When to use Notice and why can't we use Error log

Reply:
Resolved